### ISSUE TITLE
ValueError: cannot assign slice from input of different size

### ISSUE THREAD

**Issue by ghost** [pysal/esda#193]
I am using `Moran_Local` function from `esda.moran` on weight matrix derived from `Queen` contiguity on a `geodataframe`. My code is:

```
import geopandas as gpd
from pysal.explore import esda  
from pysal.lib import weights
db = gpd.read_file('file.geojson')
w_queen = weights.Queen.from_dataframe(db)
lisa = esda.moran.Moran_Local(db['attribute'], w_queen)
```
I'm getting following error:

`ValueError: cannot assign slice from input of different size`

**Comment by jGaboardi** [comment]
The OP's account seems to have been [deleted](https://stackoverflow.com/a/64984367/8692996).

**Comment by KaixuanDai** [comment]
It may be caused by the disconnect in the weight matrix. Using the distance matrix as `w=weights.distance.Kernel.from_dataframe(db['attribute'])` may be fix it.

**Comment by knaaptime** [comment]
as explained above, this failure happens because of islands in the weights matrix (duplicate of #167) . A fix is implemented in https://github.com/pysal/esda/commit/abb5568e1dcbd055dd3a9465cc0194d33353117e and will be available in the next release

**Comment by Stophface** [comment]
@knaaptime why is this closed? This is still a problem in 2.4.1

**Comment by ljwolf** [comment]
It's a duplicate of #167, which is still open.
